### PR TITLE
We should practice selflessness in this experiment.

### PR DESCRIPTION
**Pull request by mrhwick**
Changelog:
- Added a function `owner_name_added_in_diff()` which will detect the PR owner's name in the PR diff as an addition.
- Reject PRs outright that exhibit selfishness.

### CODE REVIEW

File: github_api/prs.py
Comment (ECrownofFire) [inline github_api/prs.py]: This should be case insensitive.

File: github_api/prs.py
Comment (ECrownofFire) [inline github_api/prs.py]: This should be case insensitive

Comment (mrhwick) [reply github_api/prs.py]: Good catch

**Comment by rhengles** [comment]
Why not let democracy judge that ?

**Comment by mrhwick** [comment]
That's what this PR is for, democracy is judging it as we speak.

**Comment by bperson** [comment]
@rhengles due to @PlasmaPower sneakily grabbing the power earlier :)  See #162 for a beautiful story written by @rudehn

**Comment by dehodson** [comment]
I thought about making this very same PR but decided against it, it doesn't stop anyone from just obfuscating and putting their name in. Heck, "deho" + "dson" gets around it.

**Comment by mrhwick** [comment]
Good point, but at least it makes the attack more difficult.

**Comment by PlasmaPower** [comment]
I don't like this. It creates a false sense of security. The author can just register another username or just do something like `"rewopamsalp"[::-1]`.

Edit: missed the 2 above comments

**Comment by bperson** [comment]
A possible solution is to have chaosbot edit all comments and PRs to "blank" them, leaving only the code, hence forcing everyone to look at it before voting.

**Comment by rudehn** [comment]
or make it harder to change the voting rules

**Comment by dehodson** [comment]
Or, perhaps someone other than the author needs to leave a review approving the changes? Obviously doesn't stop collusion but an army of bots mass-upvoting would also circumvent the current process.

**Comment by bperson** [comment]
@dehodson The review on #138 didn't stop people from upvoting our benevolent dictator into power.

**Comment by dehodson** [comment]
@bperson It wasn't an approval, though, and a lot of people probably voted before reading that. If there was a change made requiring someone other than the author approve the changes it would at least show that someone took the time to look at what was changed.

**Comment by bperson** [comment]
@dehodson so we should not count votes before the review? it seems super flaky to me, we might as well start voting via reviews instead of "reactions"

**Comment by PlasmaPower** [comment]
How about we require one non-author review submitted after the latest push?

Edit: and make sure the account is at least one month old.

**Comment by dehodson** [comment]
@bperson, no, no, that's not what I'm saying. I'm saying we need one approval before merging, as well as a positive vote ratio.
edit: @PlasmaPower sniped me

**Comment by bperson** [comment]
Yes, I understood that and you're not solving "a lot of people probably voted before reading that" by doing it.

It's like patching a leak with duct tape full of holes :D

**Comment by dehodson** [comment]
@bperson: The review on #138 was not an approval.

File: cron/poll_pull_requests.py
Comment (bperson) [inline cron/poll_pull_requests.py:23]: proving a point, sry :)

**Review by bperson — APPROVED** [review]

**Comment by PlasmaPower** [comment]
Maybe instead of requiring one approval, we should require more approval reviews than rejection reviews, similar to how the base voting works? The difference would be only reviews submitted after the latest push would count.

**Comment by bperson** [comment]
> we might as well start voting via reviews instead of "reactions"

:)

**Comment by PlasmaPower** [comment]
> we might as well start voting via reviews instead of "reactions"

People might, but this project relies on people doing the right thing anyway. If we make it clear that a review should only be submitted after fully reviewing the changes, it might work.

File: github_api/prs.py
Comment (reddraggone9) [inline github_api/prs.py:3]: This import doesn't appear to be used anywhere.

**Comment by xyproto** [comment]
This pull request will prevent people from accidentally submitting their usernames. Which is nice.

File: github_api/prs.py
Comment (reddraggone9) [inline github_api/prs.py:117]: This requires there be at least one character between the literal '+' and the username. You probably want `.*` instead of `.+`. It'd also be slightly more correct if you used `^` with MULTILINE.

File: cron/poll_pull_requests.py
Comment (reddraggone9) [inline cron/poll_pull_requests.py:32]: You could factor out a `reject_with_comment` function instead of duplicating this logic.

**Review by reddraggone9 — CHANGES_REQUESTED** [review]

**Comment by PlasmaPower** [comment]
This would also prevent legit PRs like #166.

**Comment by mrhwick** [comment]
I think it would be useful to require that people be honored by others as a part of legitimate PRs like #166

**Comment by rudehn** [comment]
Try this again with #153 accepted

**Comment by PlasmaPower** [comment]
I guess I got to use my power after all :D

**Pull request closed without merging**
